lua/toggleterm/terminal.lua

```diff
@@ -298,15 +305,21 @@ local function opener(size, term)
     ui.open_tab(term)
   elseif direction == "float" then
     ui.open_float(term)
+  else
+    error("Invalid direction")
```

Comment:
I prefer to use the util functions rather than `error` since error prints a pretty ugly stack trace which I find fairly distracting. Actually long term I want to migrate all errors and messages to use
```lua
vim.notify(msg, vim.log.levels.ERROR)
```

Reply:
The problem is `Terminal:open()` cannot proceed if the direction is invalid.

Reply:
The problem is `Terminal:open` cannot proceed if the direction is invalid.

Reply:
@zeertzjq hmm in which case I guess it's fine for now I'll look into a different solution (like returning an `ok` value which is a success boolean and handling things in the caller e.g. an early return) at another time since I really don't like the way it presents errors, but it's not a blocker.